Sources/Data Model/Audience/UserAttribute.swift

```diff
@@ -129,10 +136,44 @@ extension UserAttribute {
             // user attribute "less than" this condition value
             // so evaluate if this condition value "isGreater" than the user attribute value
             return try value!.isGreater(than: rawAttributeValue!)
+        case .le:
+            // user attribute "less than" or equal this condition value
+            // so evaluate if this condition value "isGreater" than or equal the user attribute value
+            return try value!.isGreaterOrEqual(than: rawAttributeValue!)
         case .gt:
             // user attribute "greater than" this condition value
             // so evaluate if this condition value "isLess" than the user attribute value
             return try value!.isLess(than: rawAttributeValue!)
+        case .ge:
+            // user attribute "greater than or equal" this condition value
+            // so evaluate if this condition value "isLess" than or equal the user attribute value
+            return try value!.isLessOrEqual(than: rawAttributeValue!)
+        // semantic versioning seems unique.  the comarison is to compare verion but the passed in version is the target version.
+        case .semver_eq:
+            guard let targetValue = AttributeValue(value: rawAttributeValue) else {
+                throw OptimizelyError.evaluateAttributeInvalidType("attribute value \(nameFinal) invalid type")
+             }
+            return try targetValue.isSemanticVersionEqual(than: value!.stringValue)
+        case .semver_lt:
+            guard let targetValue = AttributeValue(value: rawAttributeValue) else {
+                throw OptimizelyError.evaluateAttributeInvalidType("attribute value \(nameFinal) invalid type")
+             }
+            return try targetValue.isSemanticVersionLess(than: value!.stringValue)
+        case .semver_le:
+            guard let targetValue = AttributeValue(value: rawAttributeValue) else {
+                throw OptimizelyError.evaluateAttributeInvalidType("attribute value \(nameFinal) invalid type")
+             }
+            return try targetValue.isSemanticVersionLessOrEqual(than: value!.stringValue)
+        case .semver_gt:
+            guard let targetValue = AttributeValue(value: rawAttributeValue) else {
+                throw OptimizelyError.evaluateAttributeInvalidType("attribute value \(nameFinal) invalid type")
+             }
+            return try targetValue.isSemanticVersionGreater(than: value!.stringValue)
+        case .semver_ge:
+            guard let targetValue = AttributeValue(value: rawAttributeValue) else {
+                throw OptimizelyError.evaluateAttributeInvalidType("attribute value \(nameFinal) invalid type")
+             }
+            return try targetValue.isSemanticVersionGreaterOrEqual(than: value!.stringValue)
```

Comment:
We can change them (reverse compare) to be consistent with others?

```suggestion
        case .semver_eq:
            return try value!.isSemanticVersionEqual(than: rawAttributeValue!)
        case .semver_lt:
            return try value!.isSemanticVersionGreater(than: rawAttributeValue!)
        case .semver_le:
            return try value!.isSemanticVersionGreaterOrEqual(than: rawAttributeValue!)
        case .semver_gt:
            return try value!.isSemanticVersionLess(than: rawAttributeValue!)
        case .semver_ge:
            return try value!.isSemanticVersionLessOrEqual(than: rawAttributeValue!)
```

Reply:
i thought about this but i think it adds too much confusion on the semantic version.  i believe this is a better logical split even if it deviates from the others slightly.